Add support for pinning XIP SRAM (#298)
raspberrypi/picotool: README.md

`seal` allows you to sign and/or hash a binary to run on RP2350.

> By default, it will just add an embedded block to the end containing a relative load map, thus sealing the binary. The `--hash` and `--sign` arguments can be passed to include an hash and a signature; `--major`, `--minor` and `--rollback` can be passed to include a version; and `--clear` or `--pin-xip-sram` can be passed to add load map items.

Reviewer: that is confusing `--clear` or `--pin-xip-sram` can be passed to add load map items it might be technically correct, but maybe be explicit (e.g. add load map items indicating clear of RAM and pin of XRAM) but not sure if the user cares about the details

Reviewer: i guess you did something more like the first suggestion further down

Author: I've added the more explicit description

> By default, it will just add an embedded block to the end containing a relative load map, thus sealing the binary. The `--hash` and `--sign` arguments can be passed to include an hash and a signature; `--major`, `--minor` and `--rollback` can be passed to include a version; and `--clear` or `--pin-xip-sram` can be passed to add load map items indicating clear of SRAM and pin of XIP SRAM.

Reviewer 2: Grammar feels a little odd here - is this better?
```suggestion
By default, it will just add an embedded block to the end containing a relative load map, thus sealing the binary. The `--hash` and `--sign` arguments can be passed to include an hash and a signature; `--major`, `--minor` and `--rollback` can be passed to include a version; and `--clear` or `--pin-xip-sram` can be passed to add load map items indicating whether SRAM should be cleared and whether XIP SRAM should be pinned.
```

> By default, it will just add an embedded block to the end containing a relative load map, thus sealing the binary. The `--hash` and `--sign` arguments can be passed to include an hash and a signature; `--major`, `--minor` and `--rollback` can be passed to include a version; and `--clear` or `--pin-xip-sram` can be passed to add load map items indicating whether SRAM should be cleared and whether XIP SRAM should be pinned.

Reviewer 2: ```suggestion
By default, it will just add an embedded block to the end containing a relative load map, thus sealing the binary. The `--hash` and `--sign` arguments can be passed to include an hash and a signature; `--major`, `--minor` and `--rollback` can be passed to include a version; and `--clear` or `--pin-xip-sram` can be passed to add load map items indicating whether main SRAM should be cleared and whether XIP SRAM should be pinned.
```
:wink:

Author: I was not adding it here because we're not saying "all of SRAM", and there are plenty of other places in picotool & the SDK where "SRAM" means just "main SRAM"

Reviewer 2: Fair enough.

            Sign the file
        --clear
            Clear all of SRAM on load
        --pin-xip-sram

>             Pin XIP SRAM on load

Reviewer 2: Does "all of SRAM" include "XIP SRAM", or are these distinct things? (Perhaps the wording here can be made clearer?)

Author: No, they're distinct - I think I can call it "main SRAM" to be clearer

Author: I've only changed it in those option descriptions, as everywhere else doesn't say "all of" so I think just SRAM is fine
